php/PHPFileInfo/StringBasedPHPFileInfo.php

```diff
@@ -0,0 +1,456 @@
+    {
+        $interfaces = [];
+
+        if (preg_match(self::IMPLEMENTS_PATTERN, $line, $matches)) {
+            $matches = explode(',', $matches['interfaces']);
+
+            foreach ($matches as $interface) {
+                $interfaces[] = trim($interface);
+            }
+        }
+
+        return $interfaces;
+    }
+
+    private function scanLineForImports($line)
+    {
+        $imports = [];
+
+        if (strtolower(substr($line, 0, 3) == 'use')) {
+            if (preg_match(self::USE_PATTERN, $line, $use_matches) && !empty($use_matches)) {
+                $expansions = array_map('self::trim', explode(',', $use_matches['right']));
+
+                foreach ($expansions as $expansion) {
+                    if (preg_match(self::ALIAS_PATTERN, $expansion, $expansion_matches) && !empty($expansion_matches)) {
+                        $suffix = $expansion_matches['suffix'];
+
+                        if (empty($expansion_matches['alias'])) {
+                            // Get default alias
+                            $suffix_parts = explode('\\', $suffix);
+                            $alias = array_pop($suffix_parts);
+                        } else {
+                            $alias = $expansion_matches['alias'];
+                        }
+                    }
+
+                    /** empty type means import of some class **/
+                    if (empty($use_matches['type'])) {
+                        $imports[$alias] = $use_matches['left'] . $suffix;
+                    }
+                    // @todo case when $use_matches['type'] is 'constant' or 'function'
+                    // This requires change of the oputput format because
+                    // we can use the same alias string to some class and some function at once
+                }
+            }
+        }
+
+        return $imports;
+    }
+
+    private static function trim($str)
+    {
+        return trim($str, "\t\n\r\0\x0B\\ ");
+    }
+
+    public function getNamespace()
+    {
+        return $this->namespace;
+    }
+
+    public function getClass()
+    {
+        return $this->class;
+    }
+
+    public function getSuperclass()
+    {
+        return $this->superclass;
+    }
+
+    public function getInterfaces()
+    {
+        return $this->interfaces;
+    }
+
+    public function getImports()
+    {
+        return $this->imports;
+    }
+
+    public function getFullClassPath()
+    {
+        return $this->getFullPath($this->getClass());
+    }
+
+    public function hasAliasUsed($alias)
+    {
+        return !empty($this->imports[$alias]);
+    }
+
+    public function hasErrors()
+    {
+        return ! empty($this->errors);
+    }
+
+    private function addError($error)
+    {
+        $this->errors[] = $error;
+
+        return $this;
+    }
+
+    public function getErrors()
+    {
+        return $this->errors;
+    }
+
+    public function getUsedAliasesForPath($full_path)
```

Comment:
According to http://php.net/manual/en/language.namespaces.rules.php it would be not correct: 

> Fully qualified name

    This is an identifier with a namespace separator that begins with a namespace separator, such as \Foo\Bar. The namespace \Foo is also a fully qualified name.

On the other side "qualified name" may be not full path - it can be any relative path that contains namespace separator. We are interested in only those relative to root namespace.